### DISCUSSION TITLE
cannot access virtual remote

### DISCUSSION THREAD

**Discussion by f520k510** [GerryDazoo/Slinger#519] [General]
I am helping my friend to install slinger on his synology which doesn't support docker.  I have to install from ssh.
It runs well except I can't get the virtual remote to work.

Here is the log:

Version : 3.08f Running on Linux-3.10.108-armv7l-with-glibc2.4 pid= 27534 slingbox_server.py
Using config file config.ini
Connection Manager Running on port 65432 with 3 max streams using URL chao.
BOXES [('sb1', 'cable'), ('sb2', 'evpad')]
Streamer Running:  3 config.ini cable cable 65432 8388608
Checking for slingbox at ('192.168.1.178', 5211)
Streamer Running:  3 config.ini evpad evpad 65432 8388608
starting up on  port 65432 8388608
Checking for slingbox at ('192.168.1.178', 5211)
('192.168.1.178', 5211) OK
('192.168.1.178', 5211) OK
cable Using slingbox at  ('192.168.1.178', 5211)
Traceback (most recent call last):
  File "slingbox_server.py", line 1460, in <module>
evpad Using slingbox at  ('192.168.1.178', 5211)
Streamer:  cable Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
    from flask import Flask, render_template, request, render_template_string, abort, send_file
Streamer:  evpad Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
ImportError: cannot import name 'Flask' from 'flask' (unknown location)
 
 I noticed there was an import error, but I don't know what it is.
 
 Please kindly help.  Thanks.

**Comment by GerryDazoo** [comment]
You need to install the Flask and netifaces module for your Python interpreter per the ReadMe. Without Flask you won't be able to use the remote.

Which version of NAS are you using? That's what I have (DS218) my main server running on. Also note, there doesn't appears to be a usable netifaces module for the NAS so you need make sure your IP configurations for the slingboxes are correct.

**Comment by f520k510** [comment]
I installed what I believe to be flask module by running sudo pip install flask.
I didn't install netifaces becausee the installation would fail.

I will need to ask him tomorrow what version of nas he is running.  It's an old one without docker.

**Reply by GerryDazoo** [reply]
Did you install the python 3.9  NAS package? If that's the case you have two different python's on your system like me. Your sudo command installed flask on the python you're not running the server with. 
try   /path/to/the/python/your/running -m pip install Flask.

![image](https://user-images.githubusercontent.com/102599361/216838442-97e88250-1e85-4081-81f2-2dfcf97439af.png)

**Comment by f520k510** [comment]
Thanks.
It does appear that I have two versions of python on the system as I checked the package center.
So I did what you suggtested by running /bin/python3 -m pip install Flask to no avail.
Here are the commands I can think of to check:


slingbox@CHIEN-NAS:~$ which python3
/bin/python3

It does say no flask in installed.

slingbox@CHIEN-NAS:~$ flask --version
Traceback (most recent call last):
  File "/bin/flask", line 5, in <module>
    from flask.cli import main
ModuleNotFoundError: No module named 'flask.cli'


If I run this command, however, it shows I have flask installed.
slingbox@CHIEN-NAS:~$ sudo python3 -m pip show Flask
Name: Flask
Version: 2.2.2
Summary: A simple framework for building complex web applications.
Home-page: https://palletsprojects.com/p/flask
Author: Armin Ronacher
Author-email: armin.ronacher@active-4.com
License: BSD-3-Clause
Location: /usr/lib/python3.8/site-packages


Requires: click, importlib-metadata, itsdangerous, Jinja2, Werkzeug
Required-by:

If i run this command: python3 --version, it returns 3.8
I am lost.  Sorry if it is a obvious to you but I don't get it.

Thanks.

**Reply by GerryDazoo** [reply]
Don't do the sudo when you run the command. You don't need to be root and root's environment might be different from yours. 
Just /bin/python3.  

PS about it not finding the remote files. On Linux filenames are case sensitive. Also check the file permissions to make sure they are readable.   Run this command   ls -al

**Comment by f520k510** [comment]
I got it sorted out.
If I run this command: /bin/python3 -m pip install Flask, it returns some error like "pip is a package and cannot be...."
Then I run  python3 -m pip -V, and it says pip inn't installed in the first place.
I noticed there is a warning which suggests me to run python in a virtual environment, and I decided to give it a try.
I created a virtual environment and install Flask without issues.
Then I copied all the slinger files to that directory.
I then edited the sling.service to reflect the correct directory.
Then I copied the newly edited sling.service to the /etc/systemd/system/ directory, enable the service.
And all of a sudden, it works.
